docs/syntax.md

```diff
@@ -0,0 +1,340 @@
+# WireViz Syntax
+
+## Main sections
+
+```yaml
+connectors:  # dictionary of all used connectors
+  <str>   :    # unique connector designator/name
+    ...          # connector attributes (see below)
+  <str>   :
+    ...
+  ...
+
+cables:      # dictionary of all used cables and wires
+  <str>   :    # unique cable designator/name
+    ...          # cable attributes (see below)
+  <str>   :
+    ...
+  ...
+
+connections:  # list of all connections to be made
+              # between cables and connectors
+  -
+    ...         # connection set (see below)
+  -
+    ...
+  ...
+
+additional_bom_items:  # custom items to add to BOM
+  - <bom-item>           # BOM item (see below)
+  ...
+
+```
+
+## Connector attributes
+
+```yaml
+<str>   :  # unique connector designator/name
+  # general information about a connector (all optional)
+  type: <str>   
+  subtype: <str>   
```

Comment:
- Same comment as above: Are there some limitations about a string like this?
- If a _designator_ can be any string, then these descriptions can be combined.
- If the only limitations are what can be specified in YAML, then a link to a description, might be enough.